File: src/pattern/abstractFactory/listFactory/java/ListLink.java
package pattern.abstractFactory.listFactory.java;

import pattern.abstractFactory.factory.java.Link;

public class ListLink extends Link {
    public ListLink(String caption, String url){
        super(caption, url);
    }

    @Override
    public String makeHTML() {
        return "<li><a href='"+this.url+"'>"+this.caption+"</a></li>\n";
    }
}

File: src/pattern/abstractFactory/listFactory/kotlin/ListLink.kt
package pattern.abstractFactory.listFactory.kotlin

import pattern.abstractFactory.factory.kotlin.Link

class ListLink(caption: String, url: String): Link(caption, url) {
    override fun makeHTML(): String {
        return "<li><a href='${this.url}'>${this.caption}</a></li>"
    }
}